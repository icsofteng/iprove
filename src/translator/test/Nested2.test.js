const {translate} = require('../../translator/z3')
const {translate: translate_latex} = require('../../translator/latex')

const test_constants = [{ value: 'P', varType: 'Bool' }, { value: 'Q', varType: 'Bool' }, { value: 'R', varType: 'Bool' }, { value: 'T', varType: 'Bool' }]
const test_rules = [
  {
    type: 'unary',
    symbol: 'not',
    value: {
      type: 'literal',
      value: 'T'
    }
  },
  {
    type: 'binary',
    symbol: 'implies',
    lhs: {
      type: 'literal',
      value: 'P'
    },
    rhs: {
      type: 'unary',
      symbol: 'not',
      value: {
        type: 'binary',
        symbol: 'and',
        lhs: {
          type: 'literal',
          value: 'R'
        },
        rhs: {
          type: 'literal',
          value: 'Q'
        }
      }
    }
  },
  {
    type: 'binary',
    symbol: 'implies',
    lhs: {
      type: 'literal',
      value: 'P'
    },
    rhs: {
      type: 'binary',
      symbol: 'or',
      lhs: {
        type: 'literal',
        value: 'R'
      },
      rhs: {
        type: 'literal',
        value: 'T'
      }
    }
  },
  {
    type: 'binary',
    symbol: 'implies',
    lhs: {
      type: 'literal',
      value: 'P'
    },
    rhs: {
      type: 'unary',
      symbol: 'not',
      value: {
        type: 'literal',
        value: 'Q'
      }
    }
  }]

test('Nested 2 test', () => {
  expect(translate(test_rules, test_constants, [], [], [])).toMatchSnapshot()
})


test('Nested 2 test latex', () => {
  expect(translate_latex(test_rules, test_constants, [], [], [])).toMatchSnapshot()
})