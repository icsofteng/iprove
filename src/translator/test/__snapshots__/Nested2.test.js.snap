// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nested 2 test 1`] = `
"(declare-const P Bool)
(declare-const Q Bool)
(declare-const R Bool)
(declare-const T Bool)
(assert (not T))
(assert (=> P (not (and R Q))))
(assert (=> P (or R T)))
(assert (not (=> P (not Q))))
(check-sat)"
`;

exports[`Nested 2 test latex 1`] = `
Array [
  "\\\\lnot T",
  "P \\\\Longrightarrow \\\\lnot R \\\\land Q",
  "P \\\\Longrightarrow R \\\\lor T",
  "P \\\\Longrightarrow \\\\lnot Q",
]
`;
